src/SonataPageAdminTrait.php

```diff
@@ -0,0 +1,313 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the behat/helpers project.
+ *
+ * (c) Ekino
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Ekino\BehatHelpers;
+
+use Behat\Mink\Exception\ElementNotFoundException;
+
+/**
+ * @author William JEHANNE <william.jehanne@ekino.com>
+ */
+trait SonataPageAdminTrait
+{
+    /**
+     * I open the container by text.
+     *
+     * @When /^I open the container by text "([^"]*)"$/
+     *
+     * @param string $text
+     *
+     * @throws ElementNotFoundException
+     */
+    public function iOpenTheContainerByText(string $text): void
+    {
+        $element = $this->getSession()->getPage()->find('css', sprintf('div.page-composer__page-preview a:contains(\'%s\')', $text));
+
+        if (null === $element) {
+            throw new ElementNotFoundException($this->getSession()->getDriver(), null, 'text', $text);
+        }
+
+        $element->click();
+    }
+
+    /**
+     * I add a block with the name.
+     *
+     * @When /^I add the block "([^"]*)" with the name "([^"]*)"$/
+     *
+     * @param string $text
+     * @param string $name
+     *
+     * @throws ElementNotFoundException
+     */
+    public function iAddABlockWithTheName(string $text, string $name): void
+    {
+        $buttonElement = 'div.page-composer__block-type-selector button';
+        $inputElement  = sprintf('li.page-composer__container__child:contains(\'%s\') * input.page-composer__container__child__name__input', $text);
+        $blockElement  = sprintf('a.BlockSelectModal_SelectLink:contains(\'%s\')', $text);
+
+        $this->iWaitForCssElementBeingVisible($buttonElement, 2);
+
+        $button = $this->getSession()->getPage()->find('css', $buttonElement);
+
+        if (null === $button) {
+            throw new ElementNotFoundException($this->getSession()->getDriver(), null, 'text', 'Add a block');
+        }
+
+        $button->click();
+        $this->waitForSeconds(1);
+        $this->thePopinShouldBeOpened('blockSelectModal');
```

Comment:
is that possible to use clickingOnElementShouldOpenPopin instead of those 3 lines? don't know if this modal has a compatible id with this method?